athena-postgresql/src/main/java/com/amazonaws/athena/connectors/postgresql/PostGreSqlMetadataHandler.java

```diff
@@ -248,11 +250,60 @@ public GetSplitsResponse doGetSplits(BlockAllocator blockAllocator, GetSplitsReq
         return new GetSplitsResponse(getSplitsRequest.getCatalogName(), splits, null);
     }
 
+    @Override
+    protected List<TableName> listTables(final Connection jdbcConnection, final String databaseName)
+            throws SQLException
+    {
+        ImmutableList.Builder<TableName> list = ImmutableList.builder();
+
+        // Gets list of Tables and Views using Information Schema.tables
+        list.addAll(JDBCUtil.getTables(jdbcConnection, databaseName));
+        // Gets list of Materialized Views using table pg_catalog.pg_matviews 
+        list.addAll(getMaterializedViews(jdbcConnection, databaseName));
+
+        return list.build();
+    }
+
+    private String caseSensitiveNameResolver(PreparedStatement preparedStatement, String tableName, String databaseName) throws SQLException
+    {
+        String resolvedName = null;
+        try (ResultSet resultSet = preparedStatement.executeQuery()) {
+            if (resultSet.next()) {
+                resolvedName = resultSet.getString("table_name");
+                if (resultSet.next()) {
+                    throw new RuntimeException(String.format("More than one table that matches '%s' was returned from Database %s", tableName, databaseName));
+                }
+                LOGGER.info("Resolved name from Case Insensitive look up : {}", resolvedName);
+            }
+            else {
+                return null;
+            }
+        }
+        return resolvedName;
+    }
+
     @Override
     protected TableName caseInsensitiveTableSearch(Connection connection, final String databaseName,
                                                      final String tableName) throws Exception
     {
-        return JDBCUtil.informationSchemaCaseInsensitiveTableMatch(connection, databaseName, tableName);
+        return informationSchemaCaseInsensitiveTableMatch(connection, databaseName, tableName);
+    }
+
+    public TableName informationSchemaCaseInsensitiveTableMatch(Connection connection, final String databaseName,
```

Comment:
For PostGreSql, we are resolving table name from 
1. info schema
2. MV view
Can we change the name of method? so it won't collide with JDBC Utils, and make it clear the purpose of this method